File: blockchain_integration/pi_network/blockchain_integration/stellar_integration/realtime_transaction_monitoring/stellar_transaction_monitor.py
import stellar_sdk
from stellar_sdk.horizon import Horizon

class StellarTransactionMonitor:
    def __init__(self, horizon_url: str, stellar_network: str):
        self.horizon_url = horizon_url
        self.stellar_network = stellar_network
        self.horizon = Horizon(horizon_url)

    def stream_transactions(self, wallet_address: str) -> None:
        for transaction in self.horizon.transactions().for_account(wallet_address).stream():
            # Process transaction here
            print(f"Received transaction: {transaction.hash}")

    def get_transaction_history(self, wallet_address: str, limit: int = 10) -> list:
        transactions = self.horizon.transactions().for_account(wallet_address).limit(limit).call()
        return transactions

File: blockchain_integration/pi_network/blockchain_integration/stellar_integration/realtime_transaction_monitoring/stellar_transaction_monitor.go
package stellar

import (
	"context"
	"fmt"
	"log"

	"github.com/gorilla/websocket"
	"github.com/stellar/go/clients/horizon"
)

type StellarTransactionMonitor struct {
	horizonURL string
	networkPassphrase string
	client *horizon.Client
	ws *websocket.Conn
}

func NewStellarTransactionMonitor(horizonURL, networkPassphrase string) *StellarTransactionMonitor {
	return &StellarTransactionMonitor{
		horizonURL: horizonURL,
		networkPassphrase: networkPassphrase,
		client: horizon.NewClient(horizonURL),
	}
}

func (monitor *StellarTransactionMonitor) StartMonitoring() error {
	var err error
	monitor.ws, _, err = websocket.DefaultDialer.Dial(fmt.Sprintf("wss://%s/websocket", monitor.horizonURL), nil)
	if err!= nil {
		return err
	}
	err = monitor.ws.WriteMessage(websocket.TextMessage, []byte(`{"command": "listen", "stream": "transactions"}`))
	if err!= nil {
		return err
	}
	go monitor.listen()
	return nil
}

func (monitor *StellarTransactionMonitor) listen() {
	for {
		_, message, err := monitor.ws.ReadMessage()
		if err!= nil {
			log.Println(err)
			return
		}
		tx, err := txnbuild.TransactionFromXDR(message)
		if err!= nil {
			log.Println(err)
			continue
		}
		log.Println(tx.Hash)
	}
}

func (monitor *StellarTransactionMonitor) StopMonitoring() error {
	err := monitor.ws.WriteMessage(websocket.TextMessage, []byte(`{"command": "unsubscribe", "stream": "transactions"}`))
	if err!= nil {
		return err
	}
	err = monitor.ws.Close()
	return err
}
